mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -71,6 +71,22 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
 
   /** @group getParam */
   def getItemCol: String = $(itemCol)
+
+  /**
+   * Param for strategy for dealing with unknown users or items at prediction time.
+   * Supported values:
+   * - "nan": prediction value for unknown ids will be NaN.
+   * - "drop": rows in the input DataFrame containing unknown ids will be dropped.
+   * Default: "nan".
+   * @group expertParam
+   */
+  val unknownStrategy = new Param[String](this, "unknownStrategy",
```

Comment:
I personally don't see much difference between "new" and "unknown". Will
think about it some more

On Wed, 4 May 2016 at 18:08, Seth Hendrickson <notifications@github.com>
wrote:

> In mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala
> <https://github.com/apache/spark/pull/12896#discussion_r62067123>:
>
> > @@ -71,6 +71,22 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
> >
> >    /** @group getParam */
> >    def getItemCol: String = $(itemCol)
> > +
> > +  /**
> > +   * Param for strategy for dealing with unknown users or items at prediction time.
> > +   * Supported values:
> > +   * - "nan": prediction value for unknown ids will be NaN.
> > +   * - "drop": rows in the input DataFrame containing unknown ids will be dropped.
> > +   * Default: "nan".
> > +   * @group expertParam
> > +   */
> > +  val unknownStrategy = new Param[String](this, "unknownStrategy",
>
> Thinking about it more, using "unknown" just seems confusing.
> "newIdPredictionStrategy" ? I don't have a great idea for it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12896/files/fc437451a598221f0878b7a2e0b87d17572019cc#r62067123>
>